searching/binary_searching.py

```diff
@@ -0,0 +1,22 @@
+import random
+def binary_search(L, k):
```

Comment:
Nice! This is very good.
I think it would be a bit more useful, if the list contained key-value pairs. 
You could then return the actual value, eg the L[k]. If you only return the key, all this is useful for is to check if `k` is in `L` (which admittedly has it's use-cases of course).